src/Components/Server/src/Circuits/DefaultCircuitFactory.cs

```diff
@@ -3,48 +3,59 @@
 
 using System;
 using System.Linq;
+using System.Text.Encodings.Web;
 using Microsoft.AspNetCore.Components.Browser;
 using Microsoft.AspNetCore.Components.Browser.Rendering;
 using Microsoft.AspNetCore.Components.Rendering;
+using Microsoft.AspNetCore.Components.Services;
 using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
 using Microsoft.AspNetCore.SignalR;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Logging;
-using Microsoft.Extensions.Options;
+using Microsoft.JSInterop;
 
 namespace Microsoft.AspNetCore.Components.Server.Circuits
 {
     internal class DefaultCircuitFactory : CircuitFactory
     {
         private readonly IServiceScopeFactory _scopeFactory;
-        private readonly DefaultCircuitFactoryOptions _options;
         private readonly ILoggerFactory _loggerFactory;
 
         public DefaultCircuitFactory(
             IServiceScopeFactory scopeFactory,
-            IOptions<DefaultCircuitFactoryOptions> options,
             ILoggerFactory loggerFactory)
         {
-            if (options == null)
-            {
-                throw new ArgumentNullException(nameof(options));
-            }
-
             _scopeFactory = scopeFactory ?? throw new ArgumentNullException(nameof(scopeFactory));
-            _options = options.Value;
             _loggerFactory = loggerFactory;
         }
 
-        public override CircuitHost CreateCircuitHost(HttpContext httpContext, IClientProxy client)
+        public override CircuitHost CreateCircuitHost(
+            HttpContext httpContext,
+            IClientProxy client,
+            string uriAbsolute,
+            string baseUriAbsolute)
         {
-            if (!_options.StartupActions.TryGetValue(httpContext.Request.Path, out var config))
+            var metadata = ResolveComponentMetadata(httpContext, client);
+
+            var scope = _scopeFactory.CreateScope();
+            var encoder = scope.ServiceProvider.GetRequiredService<HtmlEncoder>();
+            var jsRuntime = (RemoteJSRuntime)scope.ServiceProvider.GetRequiredService<IJSRuntime>();
```

Comment:
I'm kinda partial to newing up an instance and setting it on the accessor. Both are mutating state, but it might be a bit easier to know when the accessor wasn't initialized.

Reply:
Not sure what you mean, but we can't replace one instance by another in the accessor as that won't update the references components might be holding on to it.

This is also IMO a very trivial implementation detail, there is one place in code where initialization happens (the circuit factory) and that's the important invariant to maintain.

Reply:
> Components shouldn't live past the lifetime of a single scope though. Regardless, you'd run in to the same problem since you're effectively creating instances by creating a new scope.

They don't.

> More on point, it makes sense to use the accessor pattern that's more common in our codebase. I'm all for hiding the accessor as implementation details i.e. making the accessor internal but that could be cleaned up separately. Are there technical limitations for leaving the current accessor as is?

We don't expose the accessor to the public and we don't want to / need to. Is there any real benefit on using the accessor? There is a specific issue to move away from it. The only place with an accessor is the circuithost.

Everything else belongs to the scope of the circuithost so while you are in it/use it you should be fine with plain DI.

If you feel so strongly about it, I will revert the changes, but I consider this to be a complete implementation detail.

Reply:
We can figure this offline. I wouldn't block on this now.